mixins/textInput.js

```diff
@@ -126,21 +125,13 @@ export default {
                : props.disabled ? 'minus'
                : props.icon ? props.icon
                : (props.required && !props.value) ? 'caret-left'
-               : false
-
-    if (props.onIconClick) {
-      return icon && (
-        <a href="" className={ className } onClick={ props.onIconClick }>
-          <Icon icon={ icon } fixedWidth={ true } />
-        </a>
-      )
-    } else {
-      return icon && (
-        <span className={ className }>
-          <Icon icon={ icon } fixedWidth={ true } />
-        </span>
-      )
-    }
+               : false;
+
+    return icon && (
+      <span className={ className }>
```

Comment:
Yep, I'm happy with this as a solution which doesn't have you overriding things.